rewriter/Struct.cc

```diff
@@ -21,6 +21,24 @@ bool isKeywordInitKey(const core::GlobalState &gs, const ast::ExpressionPtr &nod
     return false;
 }
 
+bool missingInitialize(const core::GlobalState &gs, const ast::Send *send) {
+    if (send->block == nullptr) {
+        return true;
+    }
+
+    auto &block = ast::cast_tree_nonnull<ast::Block>(send->block);
+
+    if (auto *insSeq = ast::cast_tree<ast::InsSeq>(block.body)) {
+        auto methodDef = ast::cast_tree<ast::MethodDef>(insSeq->expr);
+
+        if (methodDef && methodDef->name == core::Names::initialize()) {
+            return false;
+        }
+    }
```

Comment:
It seems like you'd want to check the entire `stats` vector for an appropriate `MethodDef` too?  So something like:

```ruby
Const = Struct.new(:something, keyword_init: true) do
  extend T::Sig

  sig { params(something: Integer).void }
  def initialize(something: 0)
    T.reveal_type(something) # error: Integer
  end

  def other_method
    p "just here for example purposes"
  end
end
```

should also return `false`.  IIUC, with the current code it would return `true`.

Reply:
Yes, you're right we do indeed need to go through `stats` to make sure we capture the `initialize` definition regardless of its position. The latest commit contains the changes.